Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -130,8 +130,66 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
+                                // We do not use a set in order to keep the previous children's order
+                                for child in newChildren {
+                                    if !allChildren.contains(child) {
+                                        allChildren.append(child)
+                                    }
+                                }
+
+                                let payload = mobileRoot.asPayloadWithChildren(allChildren)
+                                let mappedGUID = payload["id"].string ?? mobileRoot.guid
+                                return deferMaybe(Record<BookmarkBasePayload>(id: mappedGUID, payload: payload))
+                            }
+                        } else {
+                            let mobileRoot = BookmarkMirrorItem.folder(BookmarkRoots.MobileFolderGUID, modified: Date.now(), hasDupe: false, parentID: BookmarkRoots.RootGUID, parentName: "", title: "Mobile Bookmarks", description: nil, children: Array(newBookmarks.keys))
+                            return deferMaybe(Record<BookmarkBasePayload>(id: BookmarkRoots.translateOutgoingRootGUID(BookmarkRoots.MobileFolderGUID), payload: mobileRoot.asPayload()))
+                        }
+                    } >>== { mobileRootRecord in
+                        // Build a list of records to upload
+                        var records = Array(newBookmarks.values.map { bkm -> Record<BookmarkBasePayload> in
+                            let payload = bkm.asPayload()
+                            let mappedGUID = payload["id"].string ?? bkm.guid
+                            return Record<BookmarkBasePayload>(id: mappedGUID, payload: payload)
+                        })
+                        records.append(mobileRootRecord)
+
+                        let bufferOP = BufferUpdatedCompletionOp()
+                        let onUpload: (POSTResult, Timestamp?) -> DeferredTimestamp = { result, lastModified in
+                            let modified = lastModified ?? 0
+                            let uploaded = result.success.map { BookmarkRoots.translateIncomingRootGUID($0) }
+                            if (uploaded.contains(BookmarkRoots.MobileFolderGUID)) {
+                                bufferOP.mobileRootToUpdate = (mobileRootRecord.payload as MirrorItemable).toMirrorItem(modified)
+                            }
+                            bufferOP.bufferValuesToMoveFromLocal.formUnion(uploaded.filter { $0 != BookmarkRoots.MobileFolderGUID })
+                            bufferOP.setModifiedTime(modified, guids: uploaded)
+                            return deferMaybe(modified)
+                        }
+
+                        return self.uploadRecords(records, lastTimestamp: mirrorer.lastModified, storageClient: bookmarksClient, onUpload: onUpload) >>== { lastModified -> Success in
+                            self.setTimestamp(lastModified)
```

Comment:
Here you're switching between the mirror's `lastModified` and the synchronizer's `lastModified`.

Because you haven't bumped the mirrorer, as far as I recall next time we sync the mirrorer will download the records we just uploaded!

See this in the primary synchronizer:

```swift
                // We need to advance our batching downloader timestamp to match. See Bug 1253458.
                self.setTimestamp(timestamp)
                mirrorer.advanceNextDownloadTimestampTo(timestamp: timestamp)
```